internal/dataplane/parser/translate_udproute_test.go

```diff
@@ -76,18 +107,68 @@ func TestIngressRulesFromUDPRoutes(t *testing.T) {
 			},
 		},
 		{
-			name: "single UDPRoute with multiple rules",
+			name: "multiple UDPRoute with single rule, different SectionName and Gateway in the same namespace",
+			gateways: []*gatewayapi.Gateway{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "gateway-1",
+						Namespace: "default",
+					},
+					Spec: gatewayapi.GatewaySpec{
+						Listeners: []gatewayapi.Listener{
+							builder.NewListener("udp80").WithPort(80).UDP().Build(),
+							builder.NewListener("udp81").WithPort(81).UDP().Build(),
+						},
+					},
+				},
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "gateway-1",
+						Namespace: "test-1",
+					},
+				},
+			},
 			udpRoutes: []*gatewayapi.UDPRoute{
 				{
-					TypeMeta:   udpRouteTypeMeta,
-					ObjectMeta: metav1.ObjectMeta{Name: "multiple-rules", Namespace: "default"},
+					TypeMeta: udpRouteTypeMeta,
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "rule-1",
```

Comment:
nit: wouldn't it be better/more readable to call those something like `udproute-rule-1` just to make the expected routes more readable?

Reply:
Maybe, but on the other hand, adding `udproute` seems redundant, because it's a route in`UDPRoute` object, so how can be for different protocol